drivers/sensor/bmp388/bmp388.c

```diff
@@ -0,0 +1,691 @@
+/* Bosch BMP388 pressure sensor
+ *
+ * Copyright (c) 2020 Facebook, Inc. and its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmp388-ds001.pdf
+ */
+
+#define DT_DRV_COMPAT bosch_bmp388
+
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include "bmp388.h"
+
+LOG_MODULE_REGISTER(BMP388, CONFIG_SENSOR_LOG_LEVEL);
+
+static const struct {
+	uint16_t freq_int;
+	uint16_t freq_milli;
+} bmp388_odr_map[] = {
+	{ 0, 3 },       /* 25/8192 - 327.68s */
+	{ 0, 6 },       /* 25/4096 - 163.84s */
+	{ 0, 12 },      /* 25/2048 - 81.92s */
+	{ 0, 24 },      /* 25/1024 - 40.96s */
+	{ 0, 49 },      /* 25/512 - 20.48s */
+	{ 0, 98 },      /* 25/256 - 10.24s */
+	{ 0, 195 },     /* 25/128 - 5.12s */
+	{ 0, 391 },     /* 25/64 - 2.56s */
+	{ 0, 781 },     /* 25/32 - 1.28s */
+	{ 1, 563 },     /* 25/16 - 640ms */
+	{ 3, 125 },     /* 25/8 - 320ms */
+	{ 6, 250 },     /* 25/4 - 160ms */
+	{ 12, 500 },    /* 25/2 - 80ms */
+	{ 25, 0 },      /* 25 - 40ms */
+	{ 50, 0 },      /* 50 - 20ms */
+	{ 100, 0 },     /* 100 - 10ms */
+	{ 200, 0 },     /* 200 - 5ms */
+};
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+static int bmp388_transceive(const struct device *dev,
+			     void *data, size_t length)
+{
+	struct bmp388_data *bmp388 = DEV_DATA(dev);
+	const struct bmp388_config *cfg = DEV_CFG(dev);
+	const struct spi_buf buf = { .buf = data, .len = length };
+	const struct spi_buf_set s = { .buffers = &buf, .count = 1 };
+
+	return spi_transceive(bmp388->bus, &cfg->spi_cfg, &s, &s);
```

Comment:
Same buffers for TX and RX? Are you sure? I can only find one other instance of that in the tree, added in 810920f8d553717f7175fd8e73e571a41f136d8e. @tbursztyka are the semantics here well-defined?

Reply:
The BMI160 driver uses the same buffers as well. This is a common thing to do with SPI.

Reply:
> The BMI160 driver uses the same buffers as well.

The same `spi_buf`, yes, but not the same `spi_buf_set` as is being done here.

I'd still like a clarification from @tbursztyka on whether using the same `spi_buf_set` for both TX and RX allowed by the SPI API. Would be good to get it documented one way or the other.